swim/update_rollup_test.go

```diff
@@ -99,11 +100,23 @@ func (s *RollupTestSuite) TestFlushesAfterInterval() {
 	})
 
 	s.NotNil(s.r.FlushTimer(), "expected timer to be set")
+
+	// This is a race condition which never got triggered yet. If the scheduler
+	// decides to yield at this comment and proceed with flushing the buffer,
+	// assertion below will fail. Oops.
 	s.Len(s.r.Buffer(), 2, "expected two updates to be in buffer")
 
-	time.Sleep(2 * time.Millisecond)
+	// Flushing the buffer used to take <2ms on an otherwise idle machine. On
```

Comment:
The purpose of update_rollup.go is to collect membership updates and report them to the logging system after a period of at least X (`flushInterval`) milliseconds of "quite time" (aka the cluster convering).

I think this test is making sure that after 1ms passes, the "buffer is flushed."

The contract is not that strict. I'm not sure what type of promises Go's timers make. I'd imagine it something like "sometime after, but not earlier than the time you specify." That is OK enough for update_rollup.go too.